### PR TITLE
Use setup-ocaml v2

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

**Comment by mgree** [comment]
Thank you for this (and https://github.com/mgree/libdash/pull/14)!

This is _almost_ the right behavior, but the pin of the local `libdash` module was done on purpose. Sometimes I need to make simultaneous changes to libdash and smoosh, and pinning lets me use an as-yet-unpublished version of libdash with smoosh.

Can you please add back the pin, and then we can try it in CI?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -27,28 +28,22 @@ jobs:
...
           submodules: recursive
 
-      - name: Use OCaml ${{ matrix.ocaml-version }}
-        uses: avsm/setup-ocaml@v2
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
         with:
-          ocaml-version: ${{ matrix.ocaml-version }}
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          opam-local-packages: libdash/*.opam
```
Comment (smorimoto) [inline .github/workflows/build.yml:35]: This ensures that the setup-ocaml auto-pin and auto-depext functionalities works properly in the directory.
In fact, as you can see in this log, it installed the dev version correctly.
https://github.com/smorimoto/smoosh/runs/5330079707?check_suite_focus=true#step:4:57

Comment (mgree) [reply .github/workflows/build.yml:35]: 🤯

**Comment by mgree** [comment]
I didn't notice that... what a cool feature! Thank for well and truly improving the CI script, I appreciate it. 😁

**Comment by smorimoto** [comment]
Anytime!

**Pull request merged**